### DISCUSSION TITLE
Teardown race condition when running tests under `dbus-run-session`?

### DISCUSSION THREAD

**Discussion by xsdg** [flatpak/xdg-desktop-portal#1494] [Questions]
Howdy, y'all.  I'm one of the co-maintainers for Geeqie.  Recently, one of the other co-maintainers has been working on adding GTK4 support, which required some modifications to how we run our functional integration tests.

We found that without `xdg-desktop-portal` installed, our pattern of creating a tmpdir homedir, using `dbus-run-session`, and then immediately doing `rm -rf ${TEMP_HOMEDIR}` works fine.  But if `xdg-desktop-portal` _is_ installed, then the teardown will reliably fail with what feels like a filesystem-based race condition.  See https://github.com/BestImageViewer/geeqie/issues/1532

Any thoughts on what might be happening, or why the installation of `xdg-desktop-portal` might be affecting things?  My expectation would be that after `dbus-run-session` exits, nothing else should be touching the XDG directories.

**Comment by swick** [comment]
dbus-run-session going away doesn't necessarily mean that everything else has been terminated. It might make sense to try to figure out what remains and why.
